src/main/java/gameframework/particles/ParticleEmitter.java

```diff
@@ -0,0 +1,106 @@
+package gameframework.particles;
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Rectangle;
+import java.util.LinkedList;
+import java.util.List;
+
+import gameframework.particles.behaviors.ParticleBehavior;
+import gameframework.game.GameEntity;
+
+/**
+ * A game entity that can be added to a game universe
+ * This entity can emit, draw and update an infinite number of particles
+ */
+public class ParticleEmitter implements GameEntity {
+	/* List containing every "alive" particles that will be updated */
+	protected List<Particle> particles;
+	/* List containing every particles that should be removed */
+	protected List<Particle> deadParticles;
+	
+	/**
+	 * Constructor
+	 */
+	public ParticleEmitter() {
+		particles = new LinkedList<>();
+		deadParticles = new LinkedList<>();
+	}
+	
+	/**
+	 * Update a particle according to its behavior
+	 * @param particle Particle to update
+	 */
+	protected void update(Particle particle) {
+		particle.setTimeAlive(particle.getTimeAlive() + 1);
+		
+		if (particle.getBehavior().isDead(particle)) {
+			deadParticles.add(particle);
+		}
+		
+		if (particle.getBehavior().isDrawn(particle)) {
+			particle.getBehavior().update(particle);
+		}
+	}
+	
+	/**
+	 * Remove every dead particles contained in deadParticles
+	 */
+	protected void removeDeadParticles() {
+		for (Particle p : deadParticles) {
+			particles.remove(p);
+		}
+		deadParticles.clear();
+	}
+	
+	/**
+	 * Draw a single particle on a Graphics
+	 * @param p The particle to draw
+	 * @param g The Graphics
+	 */
+	public void drawParticle(Particle p, Graphics g) {
+		if (p.getBehavior().isDrawn(p)) {
```

Comment:
You can realize that this implementation has a problem because it doesn't use any instance variable. That's a code smell. It indicates that the code is probably not in the right place. You could replace it with `p.draw(g)` for example, but you probably don't need this method at all.

Please also avoid `p` and `g` as variable names